src/NCronJob/Registry/JobRun.cs

```diff
@@ -1,41 +1,100 @@
+using System.Diagnostics;
+
 namespace NCronJob;
 
-internal sealed class JobRun
+internal class JobRun
 {
     private int jobExecutionCount;
 
     internal JobPriority Priority { get; set; } = JobPriority.Normal;
 
     public required Guid JobRunId { get; init; }
-
     public required JobDefinition JobDefinition { get; init; }
-
     public Guid CorrelationId { get; set; } = Guid.NewGuid();
-
     public CancellationToken CancellationToken { get; set; }
+    public DateTimeOffset? RunAt { get; set; }
 
+    /// <summary>
+    /// At the moment of processing, if the difference between the current time and the scheduled time exceeds the
+    /// expiration period (grace period), the job is considered expired and should not be processed. Because the job is not processed,
+    /// but it has been dequeued then essentially the job is dropped.
+    /// </summary>
+    public TimeSpan Expiry { get; set; } = TimeSpan.FromMinutes(10);
+    public bool IsExpired(TimeProvider timeProvider) => RunAt.HasValue && timeProvider.GetUtcNow() - RunAt.Value > Expiry;
+    public bool IsOneTimeJob { get; set; }
     public object? Parameter { get; init; }
-
     public object? ParentOutput { get; set; }
-
     public int JobExecutionCount => Interlocked.CompareExchange(ref jobExecutionCount, 0, 0);
-
     public void IncrementJobExecutionCount() => Interlocked.Increment(ref jobExecutionCount);
 
     public static JobRun Create(JobDefinition jobDefinition) =>
-        new()
+        new JobRun
         {
             JobRunId = Guid.NewGuid(),
             JobDefinition = jobDefinition,
             Parameter = jobDefinition.Parameter
-        };
+        }.Initialize();
 
     public static JobRun Create(JobDefinition jobDefinition, object? parameter, CancellationToken token) =>
-        new()
+        new JobRun
         {
             JobRunId = Guid.NewGuid(),
             JobDefinition = jobDefinition,
             Parameter = parameter,
             CancellationToken = token
+        }.Initialize();
```

Comment:
Can't we move this logic to the c'tor? 
The API surface doesn't feel right.

Reply:
Done